src/server/saved_objects/migrations/README.md

```diff
@@ -0,0 +1,115 @@
+# Migrations
+
+This is the system that manages the versioning of saved object indices. It manages changes to the index, including mappings, index templates, and changes to the shape of documents from one version of Kibana to another.
+
+## Outstanding questions
+
+- Should we migrate all docs in the index between minor versions?
+  - @epixa does not want to migrate the index on minor versions
+  - @clintongormley [seems to think we should](https://github.com/elastic/kibana/issues/15100#issuecomment-398041512)
+
+- How should we disallow breaking changes to data between minor versions?
```

Comment:
I think we just have the policy in Core, and support it with testing.  If a user is using an unsupported plugin and that plugin decides to break bwc, it's their fault, not ours.